### PR TITLE
Make pins 2

### PR DESCRIPTION
**Pull request by NikhitaR-IFX**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
Integrating pins refactor and make_pins feature

### CODE REVIEW

File: ports/psoc6/modules/machine/machine_pin_phy.c [hunk 1]
```diff
@@ -2,7 +2,9 @@
 #include "py/mphal.h"
 #include "machine_pin_phy.h"
 
-machine_pin_phy_obj_t machine_pin_phy_obj[] = {
+// extern const uint8_t machine_pin_num_of_cpu_pins;
```
Comment (jaenrig-ifx) [inline ports/psoc6/modules/machine/machine_pin_phy.c:5]: This can be removed or?

File: ports/psoc6/modules/machine/machine_pin.c [hunk 2]
```diff
@@ -21,13 +21,13 @@ typedef struct _machine_pin_io_obj_t {
     uint8_t pull;
 } machine_pin_io_obj_t;
 
-#define MAX_PIN_IO MP_ARRAY_SIZE(machine_pin_phy_obj)
-machine_pin_io_obj_t *pin_io[MAX_PIN_IO] = {NULL};
+
+machine_pin_io_obj_t *pin_io[MAX_IO_PINS] = {NULL};
```
Comment (jaenrig-ifx) [inline ports/psoc6/modules/machine/machine_pin.c:25]: mmm MAX_IO_PINS is gone or=?

File: ports/psoc6/modules/machine/pins.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
+// std includes
+#include <stdlib.h>
+
```
Comment (jaenrig-ifx) [inline ports/psoc6/modules/machine/pins.c:3]: This file should not be required.

**Review by jaenrig-ifx — CHANGES_REQUESTED** [review]

File: ports/psoc6/modules/machine/machine_pin.c (hunk 2 above)
Comment (NikhitaR-IFX) [reply ports/psoc6/modules/machine/machine_pin.c:25]: MAX_IO_PINS is now defined in generated sources for boards. Will be in build.

File: ports/psoc6/modules/machine/machine_pin_phy.c (hunk 1 above)
Comment (NikhitaR-IFX) [reply ports/psoc6/modules/machine/machine_pin_phy.c:5]: Yep... done

**Review by jaenrig-ifx — APPROVED** [review]
🥳

**Pull request merged**